test/Errors.test.js

```diff
@@ -1,363 +1,488 @@
+						}
+					]
 				}
-			)
-		]);
+			})
+		).resolves.toMatchInlineSnapshot(`
+Object {
+  errors: Array [
+    ./entry-point.js
+Module Error (from ./irregular-error-loader.js):
+(Emitted value instead of an instance of Error) null,
+    ./entry-point.js
+Module Error (from ./irregular-error-loader.js):
+Error,
+    ./entry-point.js
+Module build failed (from ./irregular-error-loader.js):
+NonErrorEmittedError: (Emitted value instead of an instance of Error) a string error
+    at runLoaders (lib/NormalModule.js:379:13)
+    at node_modules/loader-runner/lib/LoaderRunner.js:364:11
+    at node_modules/loader-runner/lib/LoaderRunner.js:230:18
+    at runSyncOrAsync (node_modules/loader-runner/lib/LoaderRunner.js:143:3)
+    at iterateNormalLoaders (node_modules/loader-runner/lib/LoaderRunner.js:229:2)
+    at Array.<anonymous> (node_modules/loader-runner/lib/LoaderRunner.js:202:4)
+    at Storage.finished (node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:43:16)
+    at provider (node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:79:9)
+    at node_modules/graceful-fs/graceful-fs.js:78:16
+    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:511:3),
+  ],
+  warnings: Array [
+    ./entry-point.js
+Module Warning (from ./irregular-error-loader.js):
+(Emitted value instead of an instance of Error) null,
+    ./entry-point.js
+Module Warning (from ./irregular-error-loader.js):
+Error,
+  ],
+}
+`);
 	});
-	it("should throw a build error if no source be returned after run loaders", done => {
-		getErrors(
-			{
+	it("should match", async () => {
+		await expect(
+			compile({
 				mode: "development",
-				entry: path.resolve(base, "./no-return-loader") + "!./entry-point.js"
-			},
-			(errors, warnings) => {
-				expect(errors).toHaveLength(1);
-				const messages = errors[0].split("\n");
-				expect(messages[1]).toMatch(
-					/^Module build failed: Error: Final loader \(.+\) didn't return a Buffer or String/
-				);
-				done();
-			}
-		);
+				entry: "./no-return-loader!./entry-point.js"
+			})
+		).resolves.toMatchInlineSnapshot(`
+Object {
+  errors: Array [
+    ./entry-point.js (./no-return-loader.js!./entry-point.js)
+Module build failed: Error: Final loader (./no-return-loader.js) didn't return a Buffer or String
+    at runLoaders (lib/NormalModule.js:399:18)
+    at node_modules/loader-runner/lib/LoaderRunner.js:370:3
+    at iterateNormalLoaders (node_modules/loader-runner/lib/LoaderRunner.js:211:10)
+    at iterateNormalLoaders (node_modules/loader-runner/lib/LoaderRunner.js:218:10)
+    at node_modules/loader-runner/lib/LoaderRunner.js:233:3
+    at runSyncOrAsync (node_modules/loader-runner/lib/LoaderRunner.js:124:12)
+    at iterateNormalLoaders (node_modules/loader-runner/lib/LoaderRunner.js:229:2)
+    at Array.<anonymous> (node_modules/loader-runner/lib/LoaderRunner.js:202:4)
+    at Storage.finished (node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:43:16)
```

Comment:
These stacktraces depend on a lot of factors. It's a bit of a race condition what's the exact stack trace. Is the resource cached or not? How fast was the fs access? What's the last operation?

I think it's better to exclude them from the snapshot, i. e. replace them with a `STACK TRACE` label.